src/sinks/logdna.rs

```diff
@@ -0,0 +1,285 @@
+use crate::{
+    event::{self, Event},
+    sinks::util::http::{Auth, BatchedHttpSink, HttpSink},
+    sinks::util::{BatchBytesConfig, BoxedRawValue, JsonArrayBuffer, TowerRequestConfig, UriSerde},
+    topology::config::{DataType, SinkConfig, SinkContext, SinkDescription},
+};
+use http::{Request, Uri};
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use std::time::SystemTime;
+
+lazy_static::lazy_static! {
+    static ref HOST: UriSerde = Uri::from_static("https://logs.logdna.com/logs/ingest").into();
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct LogdnaConfig {
+    api_key: String,
+    host: Option<UriSerde>,
+
+    hostname: String,
+    mac: Option<String>,
+    ip: Option<String>,
+    tags: Option<Vec<String>>,
+
+    default_app: Option<String>,
+
+    #[serde(default)]
+    batch: BatchBytesConfig,
+
+    #[serde(default)]
+    request: TowerRequestConfig,
+}
+
+inventory::submit! {
+    SinkDescription::new_without_default::<LogdnaConfig>("logdna")
+}
+
+#[typetag::serde(name = "logdna")]
+impl SinkConfig for LogdnaConfig {
+    fn build(&self, cx: SinkContext) -> crate::Result<(super::RouterSink, super::Healthcheck)> {
+        let request_settings = self.request.unwrap_with(&TowerRequestConfig::default());
+        let batch_settings = self.batch.unwrap_or(bytesize::mib(10u64), 1);
+
+        let sink = BatchedHttpSink::new(
+            self.clone(),
+            JsonArrayBuffer::default(),
+            request_settings,
+            batch_settings,
+            None,
+            &cx,
+        );
+
+        let healthcheck = Box::new(futures::future::ok(()));
```

Comment:
So this is a null healthcheck? Is this because we'll know when the service fails because a push will fail?

Reply:
Correct, https://github.com/timberio/vector/blob/master/src/sinks/new_relic_logs.rs#L96 we do this in our other sinks where its a known downstream service that is hosted and not something you would deploy.

Reply:
When did we decide that healthchecks were not relevant for services that are hosted? It seems like a simple check to verify the API key authenticity would make sense here.

Reply:
Not sure! I don't think we had a clear distinction either way. 

It looks like datadog actually exposes a validate endpoint so that makes sense to make a healthcheck.